File: ChatGPT/Patrón de Diseño Singleton/c++/singleton.cpp
#include <iostream>
class Singleton {
    static Singleton* instance;
    Singleton() {}
public:
    static Singleton* getInstance() {
        if (!instance)
            instance = new Singleton();
        return instance;
    }
};
Singleton* Singleton::instance = nullptr;

int main() {
    Singleton* s1 = Singleton::getInstance();
    Singleton* s2 = Singleton::getInstance();
    std::cout << "Puntero s1: " << s1 << std::endl;
    std::cout << "Puntero s2: " << s2 << std::endl;
    std::cout << "¿Misma instancia? " << (s1 == s2 ? "sí" : "no") << std::endl;
}

File: ChatGPT/Patrón de Diseño Singleton/javascript/singleton.js
class Singleton {
    constructor() {
        if (Singleton.instance) return Singleton.instance;
        Singleton.instance = this;
    }
    static getInstance() {
        if (!Singleton.instance) Singleton.instance = new Singleton();
        return Singleton.instance;
    }
}

const s1 = Singleton.getInstance();
const s2 = Singleton.getInstance();
console.log("¿Misma instancia?", s1 === s2);
